cgsmiles/graph_layout.py

```diff
@@ -199,24 +275,78 @@ def vsepr_layout(self):
                     dist[source][dest] =  d / 2 * math.sqrt(3)
         # do we need this try except????
         try:
-            p0 = nx.fruchterman_reingold_layout(self.graph)
+            p0 = nx.fruchterman_reingold_layout(self.graph, iterations=100)
         except nx.NetworkXException:
             p0 = None
 
         pos = nx.kamada_kawai_layout(self.graph,
                                      pos=p0,
                                      weight=None,
                                      dist=dist,
-                                     scale=len(self.graph) ** 0.5)
+                                     scale=1.0)
+        avg_dist = 0
+        for edge in self.graph.edges:
+            avg_dist += np.linalg.norm(pos[edge[0]]-pos[edge[1]])
+        avg_dist = avg_dist / len(self.graph.edges)
+
+        for node in pos:
+            pos[node] *= self.default_bond / avg_dist
         return pos
 
+    def circular(self):
+        positions = generate_circle_coordinates(radius=self.default_bond*2,
+                                                num_points=len(self.graph))
+
+        pos = {}
+        start = list(self.graph.nodes)[0]
+        for idx, (node, next_node) in enumerate(nx.find_cycle(self.graph, source=start)):
```

Comment:
Does this find all cycles or only the first?

Reply:
it find a cyclic path. In this case we assume that users use it to draw a cycle